search/v2/geocoder.cpp

```diff
@@ -609,6 +659,44 @@ void Geocoder::FillLocalitiesTable(MwmContext const & context)
   }
 }
 
+void Geocoder::FillVillageLocalities()
+{
+  vector<Locality> preLocalities;
+  FillLocalityCandidates(m_villages, kMaxNumVillages, preLocalities);
+
+  size_t numVillages = 0;
+
+  for (auto & l : preLocalities)
+  {
+    FeatureType ft;
+    m_context->m_vector.GetByIndex(l.m_featureId, ft);
+
+    switch (m_model.GetSearchType(ft))
+    {
+    case SearchModel::SEARCH_TYPE_CITY:
+    {
+      if (numVillages < kMaxNumVillages && ft.GetFeatureType() == feature::GEOM_POINT)
+      {
+        ++numVillages;
+        City village = l;
+        village.m_rect = MercatorBounds::RectByCenterXYAndSizeInMeters(
+            ft.GetCenter(), ftypes::GetRadiusByPopulation(ft.GetPopulation()));
+
+#if defined(DEBUG)
+        string name;
+        ft.GetName(StringUtf8Multilang::DEFAULT_CODE, name);
+        LOG(LDEBUG, ("Village =", name));
+#endif
+
+        m_cities[{l.m_startToken, l.m_endToken}].push_back(village);
```

Comment:
m_cities are filled with cities only once (from World.mwm), but villages are loaded from each mwm separately. Probably we need to remove villages from m_cities when we're moving from one mwm to another?